engine.py

```diff
@@ -322,16 +471,27 @@ def main(args):
     Kicks off the engine with the given arguments.
     """
     try:
-        eng = Engine(args["server_port"],
-                     args["strategy"],
-                     environment_id=args.get("environment_id"),
-                     output_directory = args.get("output_directory"),
-                     log_level=args["log"])
-        eng.initialize_nfqueue()
+        if WINDOWS:
+            eng = WindowsEngine(args["server_port"],
```

Comment:
what are your thoughts on making this more generic, so a consumer would not need to know these details? the engine internally would discover the OS and then create the correct version of the engine under the hood.

Reply:
That's what I was aiming for, I think I just couldn't wrap my head around it for some reason.

I'm really looking for an interface or something to use, but python doesn't have those. I know it has ABC, I might try that.

Regardless, I'm considering Windows stuff pre-alpha atm, until we work out these few things and get the tests going.